lib/gui/components/safe-webview/safe-webview.js

```diff
@@ -0,0 +1,231 @@
+/*
+ * Copyright 2017 resin.io
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+'use strict';
+
+const _ = require('lodash');
+const electron = require('electron');
+const angular = require('angular');
+const react = require('react');
+const propTypes = require('prop-types');
+const react2angular = require('react2angular').react2angular;
+const analytics = require('../../modules/analytics');
+const packageJSON = require('../../../../package.json');
+
+const MODULE_NAME = 'Etcher.Components.SafeWebview';
+const angularSafeWebview = angular.module(MODULE_NAME, []);
+
+/**
+ * @summary Webviews that time out
+ * @type {Object}
+ * @public
+ */
+class SafeWebview extends react.PureComponent {
+
+  /**
+   * @param {Object} props - React element properties
```

Comment:
The `propTypes` section specifies the types so maybe it's better to do there? But yes we should add descriptive comments for the properties.